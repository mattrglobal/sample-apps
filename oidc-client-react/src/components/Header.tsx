import * as React from 'react';

export interface IHeaderProps {
  pageTitle: string;
  logoSrc: any;
}

export default function Header(props: IHeaderProps) {
  return (
    <header className="App-header">
      <img src={props.logoSrc} className="App-logo" alt="logo" />
      <h1 className="App-title">{props.pageTitle}</h1>
    </header>
  );
}
