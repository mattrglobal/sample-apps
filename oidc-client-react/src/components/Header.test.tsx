import { shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import logo from '../logo.svg';
import Header from './Header';

describe(Header, () => {
  const pageTitle = 'React app with oidc-client-js';

  const component = shallow(<Header pageTitle={pageTitle} logoSrc={logo} />);

  it('renders and matches our snapshot', () => {
    const headerComponent = renderer.create(<Header pageTitle={pageTitle} logoSrc={logo} />);
    const tree = headerComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the image and h1', () => {
    expect(component.find('img')).toHaveLength(1);
    expect(component.find('h1')).toHaveLength(1);
  });

  it('h1 contains correct title', () => {
    expect(component.find('h1').text()).toEqual(pageTitle);
    expect(component.contains(pageTitle)).toEqual(true);
  });
});
