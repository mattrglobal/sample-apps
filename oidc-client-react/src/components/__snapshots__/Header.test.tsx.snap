// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders and matches our snapshot 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <h1
    className="App-title"
  >
    React app with oidc-client-js
  </h1>
</header>
`;
